src/SprykerEco/Zed/Heidelpay/HeidelpayConfig.php

```diff
@@ -132,23 +134,22 @@ public function getEncryptionKey()
      */
     public function getMerchantTransactionChannelByPaymentType($paymentType)
     {
-        switch ($paymentType) {
-            case SharedHeidelpayConfig::PAYMENT_METHOD_SOFORT:
-                return $this->get(HeidelpayConstants::CONFIG_HEIDELPAY_TRANSACTION_CHANNEL_SOFORT);
-
-            case SharedHeidelpayConfig::PAYMENT_METHOD_CREDIT_CARD_SECURE:
-                return $this->get(HeidelpayConstants::CONFIG_HEIDELPAY_TRANSACTION_CHANNEL_CC_3D_SECURE);
+        if ($paymentType === SharedHeidelpayConfig::PAYMENT_METHOD_SOFORT) {
```

Comment:
I would create configuration with map:
```SharedHeidelpayConfig::PAYMENT_METHOD_SOFORT => HeidelpayConstants::CONFIG_HEIDELPAY_TRANSACTION_CHANNEL_SOFORT```
and then use lookup in it and corresponding call.

It will make method short, and configuration could be easily extended.

Reply:
Done